dask/array/routines.py

```diff
@@ -411,11 +411,24 @@ def round(a, decimals=0):
 
 
 @wraps(np.unique)
-def unique(x):
+def unique(x, return_index=False):
+    def unique(x):
+        return np.unique(x, return_index=return_index)
+
     name = 'unique-' + x.name
-    dsk = dict(((name, i), (np.unique, key)) for i, key in enumerate(x._keys()))
+    dsk = dict(((name, i), (unique, key))
```

Comment:
Can we change this to `(np.unique, key, return_index)`?  This makes the graph slightly easier to serialize and inspect.  It's a little prone to error if numpy changes the keyword order in the future.  A guard against this would be `(apply, np.unique, [key], {'return_index': return_index})` although this is obviously less pretty.